### ISSUE TITLE
Timeout when using ping3 with threading

### ISSUE THREAD

**Issue by GoopyAspirin** [kyan001/ping3#15]
Situation A) I use threading. Attempt to ping multiple servers at the same time.
(Ping is sent using threading to about 20 hosts.)
(This problem also occurs if you add a delay with time.sleep. If there are more than two pings, you will have problems.)

This problem occurs with the most recently released version.
If I use an older version, it works fine even if I send a Ping at the same time.

However, the older version has a problem that no value is returned for 5 minutes when a Timeout occurs. Therefore I can not use the old version.

Problem: Sending more than one ping at the same time or at the same time returns a timeout value.
In fact, the target host normally receives a ping.

My execution environment: I use threading. I use sleep.time.
Continue to ping about 20 hosts.

I add threading every three seconds. And it keeps running without stopping.
And that threading adds up to 30.

In the process, even though the target server is OK, Ping3 returns me a Timeout. If I ping directly from os, the response is fine.

I sincerely hope that Ping3 will be the representative package in Python 3.
I hope you can solve this problem.
Thank you.

**Comment by GoopyAspirin** [comment]
```
[2020-01-15 02:13:53] : PING :: 1.0.0.1 , 5ms
[2020-01-15 02:13:53] : PING :: 1.1.1.1 , 6ms
[2020-01-15 02:13:53] : PING :: 1.0.0.1 , 6ms
[2020-01-15 02:13:53] : PING :: 1.0.0.1 , 5ms
[2020-01-15 02:13:53] : PING :: 1.1.1.1 , 5ms
[2020-01-15 02:13:53] : PING :: 1.1.1.1 , 7ms
[2020-01-15 02:13:53] : PING :: 1.0.0.1 , 6ms
[2020-01-15 02:13:53] : PING :: 1.1.1.1 , 5ms
[2020-01-15 02:13:53] : PING :: 1.0.0.1 , 5ms
[2020-01-15 02:13:53] : PING :: 1.1.1.1 , 5ms
[2020-01-15 02:13:53] : PING :: 1.1.1.1 , 5ms
[2020-01-15 02:13:53] : PING :: 1.0.0.1 , 6ms
[2020-01-15 02:13:53] : PING :: 1.0.0.1 , 4ms
[2020-01-15 02:13:53] : PING :: 1.0.0.1 , 5ms
[2020-01-15 02:13:53] : PING :: 1.1.1.1 , 5ms
[2020-01-15 02:13:53] : PING :: 1.1.1.1 , 5ms
[2020-01-15 02:13:53] : PING :: 1.1.1.1 , 5ms
[2020-01-15 02:13:53] : PING :: 1.0.0.1 , 5ms
[2020-01-15 02:13:53] : PING :: 1.1.1.1 , 4ms
[2020-01-15 02:13:53] : PING :: 1.0.0.1 , 4ms
[2020-01-15 02:13:53] : PING :: 1.1.1.1 , 4ms
[2020-01-15 02:13:53] : PING :: 1.0.0.1 , 7ms
[2020-01-15 02:13:53] : PING :: 8.8.8.8 , 31ms
[2020-01-15 02:13:53] : PING :: 1.1.1.1 , 5ms
[2020-01-15 02:13:53] : PING :: 8.8.4.4 , 34ms
[2020-01-15 02:13:53] : PING :: 8.8.8.8 , 33ms
[2020-01-15 02:13:53] : PING :: 8.8.8.8 , 33ms
[2020-01-15 02:13:53] : PING :: 8.8.8.8 , 35ms
[2020-01-15 02:13:53] : PING :: 1.1.1.1 , 7ms
[2020-01-15 02:13:53] : PING :: 8.8.4.4 , 36ms
[2020-01-15 02:13:53] : PING :: 1.0.0.1 , 10ms
[2020-01-15 02:13:53] : PING :: 8.8.4.4 , 39ms
[2020-01-15 02:13:53] : PING :: 8.8.8.8 , 35ms
[2020-01-15 02:13:53] : PING :: 1.0.0.1 , 8ms
[2020-01-15 02:13:53] : PING :: 8.8.4.4 , 37ms
[2020-01-15 02:13:53] : PING :: 8.8.8.8 , 38ms
[2020-01-15 02:13:53] : PING :: 8.8.4.4 , 46ms
[2020-01-15 02:13:53] : PING :: 8.8.4.4 , 40ms
[2020-01-15 02:13:54] : PING :: 8.8.8.8 , 38ms
[2020-01-15 02:13:54] : PING :: 8.8.8.8 , 41ms
[2020-01-15 02:13:54] : PING :: 8.8.4.4 , 40ms
[2020-01-15 02:13:54] : PING :: 8.8.4.4 , 50ms
[2020-01-15 02:13:54] : PING :: 8.8.8.8 , 31ms
[2020-01-15 02:13:54] : PING :: 8.8.4.4 , 36ms
[2020-01-15 02:13:54] : PING :: 8.8.8.8 , 32ms
[2020-01-15 02:13:54] : PING :: 8.8.8.8 , 42ms
[2020-01-15 02:13:54] : PING :: 1.0.0.1 , 8ms
[2020-01-15 02:13:54] : PING :: 8.8.4.4 , 38ms
[2020-01-15 02:13:54] : PING :: 1.1.1.1 , 15ms
[2020-01-15 02:13:54] : PING :: 1.1.1.1 , 9ms
[2020-01-15 02:13:54] : PING :: 8.8.8.8 , 33ms
[2020-01-15 02:13:54] : PING :: 8.8.4.4 , 38ms
[2020-01-15 02:13:54] : PING :: 1.0.0.1 , 11ms
[2020-01-15 02:13:54] : PING :: 8.8.4.4 , 36ms
[2020-01-15 02:13:54] : PING :: 8.8.8.8 , 41ms
[2020-01-15 02:13:54] : PING :: 8.8.4.4 , 51ms
[2020-01-15 02:13:54] : PING :: 1.1.1.1 , 4ms
[2020-01-15 02:13:54] : PING :: 1.0.0.1 , 5ms
[2020-01-15 02:13:54] : PING :: 8.8.8.8 , 33ms
[2020-01-15 02:13:54] : PING :: 1.1.1.1 , 4ms
[2020-01-15 02:13:54] : PING :: 1.0.0.1 , 4ms
[2020-01-15 02:13:54] : PING :: 1.1.1.1 , 4ms
[2020-01-15 02:13:54] : PING :: 8.8.4.4 , 32ms
[2020-01-15 02:13:54] : PING :: 1.0.0.1 , 5ms
[2020-01-15 02:13:54] : PING :: 1.1.1.1 , 4ms
[2020-01-15 02:13:54] : PING :: 1.0.0.1 , 4ms
[2020-01-15 02:13:54] : PING :: 8.8.8.8 , 34ms
[2020-01-15 02:13:54] : PING :: 1.0.0.1 , 14ms
[2020-01-15 02:13:54] : PING :: 8.8.4.4 , 33ms
[2020-01-15 02:13:54] : PING :: 8.8.8.8 , 31ms
[2020-01-15 02:13:54] : PING :: 1.1.1.1 , 10ms
[2020-01-15 02:13:54] : PING :: 8.8.8.8 , 32ms
[2020-01-15 02:13:54] : PING :: 8.8.8.8 , 34ms
[2020-01-15 02:13:54] : PING :: 1.0.0.1 , 10ms
[2020-01-15 02:13:54] : PING :: 1.1.1.1 , 10ms
[2020-01-15 02:13:54] : PING :: 8.8.4.4 , 35ms
[2020-01-15 02:13:54] : PING :: 8.8.4.4 , 39ms
[2020-01-15 02:13:54] : PING :: 8.8.8.8 , 38ms
[2020-01-15 02:13:54] : PING :: 8.8.4.4 , 43ms
[2020-01-15 02:13:54] : PING :: 8.8.8.8 , 31ms
[2020-01-15 02:13:54] : PING :: 8.8.4.4 , 43ms
[2020-01-15 02:13:54] : PING :: 1.0.0.1 , 4ms
[2020-01-15 02:13:54] : PING :: 1.1.1.1 , 5ms
[2020-01-15 02:13:54] : PING :: 8.8.4.4 , 33ms
[2020-01-15 02:13:54] : PING :: 1.1.1.1 , 6ms
[2020-01-15 02:13:54] : PING :: 1.0.0.1 , 4ms
[2020-01-15 02:13:54] : PING :: 1.1.1.1 , 4ms
[2020-01-15 02:13:54] : PING :: 1.0.0.1 , 7ms
[2020-01-15 02:13:54] : PING :: 1.1.1.1 , 6ms
[2020-01-15 02:13:54] : PING :: 1.0.0.1 , 5ms
[2020-01-15 02:13:54] : PING :: 8.8.8.8 , 31ms
[2020-01-15 02:13:54] : PING :: 1.1.1.1 , 6ms
[2020-01-15 02:13:54] : PING :: 1.0.0.1 , 5ms
[2020-01-15 02:13:54] : PING :: 8.8.4.4 , 36ms
[2020-01-15 02:13:54] : PING :: 8.8.8.8 , 32ms
[2020-01-15 02:13:54] : PING :: 8.8.4.4 , 33ms
[2020-01-15 02:13:54] : PING :: 1.1.1.1 , 4ms
[2020-01-15 02:13:54] : PING :: 1.0.0.1 , 5ms
[2020-01-15 02:13:54] : PING :: 8.8.8.8 , 30ms
[2020-01-15 02:13:54] : PING :: 8.8.4.4 , 32ms
[2020-01-15 02:13:55] : PING :: 1.0.0.1 , 153ms
[2020-01-15 02:13:55] : PING :: 8.8.8.8 , 249ms
[2020-01-15 02:13:55] : PING :: 1.1.1.1 , 496ms
[2020-01-15 02:13:55] : PING :: 1.1.1.1 , 150ms
[2020-01-15 02:13:55] : PING :: 8.8.8.8 , 823ms
[2020-01-15 02:13:55] : PING :: 8.8.8.8 , 49ms
[2020-01-15 02:13:55] : PING :: 8.8.4.4 , 663ms
[2020-01-15 02:13:55] : PING :: 8.8.4.4 , 829ms
[2020-01-15 02:13:55] : PING :: 8.8.8.8 , 740ms
[2020-01-15 02:13:55] : PING :: 8.8.4.4 , 869ms
[2020-01-15 02:13:55] : PING :: 8.8.8.8 , 673ms
[2020-01-15 02:13:56] : PING :: 8.8.8.8 , 1117ms
[2020-01-15 02:13:56] : PING :: 1.0.0.1 , 43ms
[2020-01-15 02:13:56] : PING :: 1.0.0.1 , 225ms
[2020-01-15 02:13:56] : PING :: 8.8.8.8 , 165ms
[2020-01-15 02:13:56] : PING :: 1.1.1.1 , 171ms
[2020-01-15 02:13:56] : PING :: 8.8.4.4 , 1279ms
[2020-01-15 02:13:56] : PING :: 8.8.4.4 , 227ms
[2020-01-15 02:13:56] : PING :: 8.8.4.4 , 183ms
[2020-01-15 02:13:56] : PING :: 8.8.4.4 , 1421ms
[2020-01-15 02:13:56] : PING :: 1.0.0.1 , 129ms
[2020-01-15 02:13:56] : PING :: 1.1.1.1 , 102ms
[2020-01-15 02:13:57] : PING :: 1.1.1.1 , 87ms
[2020-01-15 02:13:57] : PING :: 8.8.4.4 , 179ms
[2020-01-15 02:13:57] : PING :: 8.8.8.8 , 149ms
[2020-01-15 02:13:57] : PING :: 1.0.0.1 , 13ms
[2020-01-15 02:13:57] : PING :: 1.1.1.1 , 31ms
[2020-01-15 02:13:57] : PING :: 8.8.8.8 , 177ms
[2020-01-15 02:13:57] : PING :: 1.0.0.1 , 71ms
[2020-01-15 02:13:57] : PING :: 1.0.0.1 , 110ms
[2020-01-15 02:13:57] : PING :: 1.1.1.1 , 341ms
[2020-01-15 02:13:57] : PING :: 8.8.8.8 , 336ms
[2020-01-15 02:13:57] : PING :: 8.8.4.4 , 545ms
[2020-01-15 02:13:57] : PING :: 1.0.0.1 , 152ms
[2020-01-15 02:13:57] : PING :: 8.8.8.8 , 131ms
[2020-01-15 02:13:57] : PING :: 8.8.4.4 , 79ms
[2020-01-15 02:13:57] : PING :: 8.8.4.4 , 464ms
[2020-01-15 02:13:57] : PING :: 1.1.1.1 , 38ms
[2020-01-15 02:13:57] : PING :: 8.8.8.8 , 60ms
[2020-01-15 02:13:58] : PING :: 8.8.4.4 , 224ms
[2020-01-15 02:13:58] : PING :: 1.1.1.1 , 82ms
[2020-01-15 02:13:58] : PING :: 1.0.0.1 , 101ms
[2020-01-15 02:13:58] : PING :: 1.1.1.1 , 65ms
[2020-01-15 02:13:58] : PING :: 1.0.0.1 , 234ms
[2020-01-15 02:13:58] : PING :: 8.8.4.4 , 270ms
[2020-01-15 02:13:58] : PING :: 8.8.8.8 , 365ms
[2020-01-15 02:13:58] : PING :: 1.1.1.1 , 126ms
[2020-01-15 02:13:58] : PING :: 8.8.8.8 , 197ms
[2020-01-15 02:13:58] : PING :: 1.0.0.1 , 152ms
[2020-01-15 02:13:58] : PING :: 8.8.8.8 , 125ms
[2020-01-15 02:13:59] : PING :: 8.8.4.4 , 337ms
[2020-01-15 02:13:59] : PING :: 8.8.4.4 , 302ms
[2020-01-15 02:13:59] : PING :: 8.8.8.8 , 63ms
[2020-01-15 02:13:59] : PING :: 1.0.0.1 , 199ms
[2020-01-15 02:13:59] : PING :: 8.8.4.4 , 91ms
[2020-01-15 02:13:59] : PING :: 1.1.1.1 , 206ms
[2020-01-15 02:13:59] : PING :: 1.1.1.1 , 117ms
[2020-01-15 02:13:59] : PING :: 1.0.0.1 , 70ms
[2020-01-15 02:13:59] : PING :: 8.8.8.8 , 124ms
[2020-01-15 02:13:59] : PING :: 1.1.1.1 , 23ms
[2020-01-15 02:13:59] : PING :: 1.1.1.1 , 5ms
[2020-01-15 02:13:59] : PING :: 1.0.0.1 , 168ms
[2020-01-15 02:13:59] : PING :: 8.8.4.4 , 303ms
[2020-01-15 02:14:00] : PING :: 1.0.0.1 , 99ms
[2020-01-15 02:14:00] : PING :: 8.8.4.4 , 249ms
[2020-01-15 02:14:00] : PING :: 8.8.8.8 , 315ms
[2020-01-15 02:14:00] : PING :: 8.8.8.8 , 355ms
[2020-01-15 02:14:00] : PING :: 1.1.1.1 , 50ms
[2020-01-15 02:14:00] : PING :: 1.0.0.1 , 169ms
[2020-01-15 02:14:00] : PING :: 8.8.8.8 , 235ms
[2020-01-15 02:14:00] : PING :: 1.0.0.1 , 60ms
[2020-01-15 02:14:00] : PING :: 8.8.4.4 , 446ms
[2020-01-15 02:14:00] : PING :: 8.8.4.4 , 176ms
[2020-01-15 02:14:00] : PING :: 1.1.1.1 , 286ms
[2020-01-15 02:14:00] : PING :: 8.8.4.4 , 134ms
[2020-01-15 02:14:00] : PING :: 1.0.0.1 , 59ms
[2020-01-15 02:14:00] : PING :: 8.8.8.8 , 95ms
[2020-01-15 02:14:00] : PING :: 1.1.1.1 , 188ms
[2020-01-15 02:14:01] : PING :: 1.1.1.1 , 37ms
[2020-01-15 02:14:01] : PING :: 8.8.8.8 , 222ms
[2020-01-15 02:14:01] : PING :: 8.8.8.8 , 92ms
[2020-01-15 02:14:01] : PING :: 1.0.0.1 , 172ms
[2020-01-15 02:14:01] : PING :: 1.0.0.1 , 184ms
[2020-01-15 02:14:01] : PING :: 8.8.8.8 , 224ms
[2020-01-15 02:14:01] : PING :: 8.8.4.4 , 479ms
[2020-01-15 02:14:01] : PING :: 8.8.4.4 , 372ms
[2020-01-15 02:14:01] : PING :: 1.0.0.1 , 163ms
[2020-01-15 02:14:01] : PING :: 1.1.1.1 , 154ms
[2020-01-15 02:14:01] : PING :: 1.1.1.1 , 483ms
[2020-01-15 02:14:01] : PING :: 8.8.4.4 , 218ms
[2020-01-15 02:14:01] : PING :: 8.8.4.4 , 39ms
[2020-01-15 02:14:01] : PING :: 8.8.8.8 , 161ms
[2020-01-15 02:14:02] : PING :: 1.0.0.1 , 43ms
[2020-01-15 02:14:02] : PING :: 1.1.1.1 , 129ms
[2020-01-15 02:14:02] : PING :: 8.8.8.8 , 99ms
[2020-01-15 02:14:02] : PING :: 1.1.1.1 , 32ms
[2020-01-15 02:14:02] : PING :: 1.0.0.1 , 174ms
[2020-01-15 02:14:02] : PING :: 8.8.4.4 , 105ms
[2020-01-15 02:14:02] : PING :: 1.0.0.1 , 195ms
[2020-01-15 02:14:02] : PING :: 8.8.8.8 , 80ms
[2020-01-15 02:14:02] : PING :: 8.8.4.4 , 211ms
[2020-01-15 02:14:02] : PING :: 8.8.4.4 , 62ms
[2020-01-15 02:14:02] : PING :: 1.1.1.1 , 97ms
[2020-01-15 02:14:02] : PING :: 8.8.8.8 , 194ms
[2020-01-15 02:14:02] : PING :: 1.1.1.1 , 332ms
[2020-01-15 02:14:02] : PING :: 1.0.0.1 , 105ms
[2020-01-15 02:14:05] : PING :: 8.8.4.4 , Request timeout
[2020-01-15 02:14:05] : PING :: 1.1.1.1 , Request timeout
[2020-01-15 02:14:05] : PING :: 8.8.8.8 , Request timeout
[2020-01-15 02:14:05] : PING :: 1.1.1.1 , Request timeout
[2020-01-15 02:14:05] : PING :: 1.0.0.1 , Request timeout
[2020-01-15 02:14:05] : PING :: 8.8.8.8 , Request timeout
[2020-01-15 02:14:05] : PING :: 8.8.8.8 , Request timeout
[2020-01-15 02:14:05] : PING :: 8.8.4.4 , Request timeout
[2020-01-15 02:14:05] : PING :: 1.0.0.1 , Request timeout
[2020-01-15 02:14:05] : PING :: 8.8.4.4 , Request timeout
[2020-01-15 02:14:05] : PING :: 1.0.0.1 , Request timeout
```

**Comment by friendlyapple** [comment]
@GoopyAspirin I am looking at where the problem could come from. In the meantime have you looked at these libraries:
- multi-ping (https://github.com/romana/multi-ping): simple library specialized in multiping
- icmplib (https://github.com/ValentinBELYN/icmplib): light and versatile library allowing ping and multiping

**Comment by kyan001** [comment]
This should be fixed in **>=v2.4.4**.
Let me know if the problem exist.
